service/controller/app/resource/chartoperator/create_test.go

```diff
@@ -0,0 +1,400 @@
+				newV1alpha1App("1abc2-cluster-autoscaler", "cluster-autoscaler", "1abc2", "org-acme", false),
+				newV1alpha1App("1abc2-kiam", "kiam", "1abc2", "org-acme", false),
+				// different cluster
+				newV1alpha1App("3def4-app-operator", "app-operator", "3def4", "org-acme", true),
+				newV1alpha1App("3def4-cert-exporter", "cert-exporter", "3def4", "org-acme", false),
+				newV1alpha1App("3def4-cert-operator", "cert-operator", "3def4", "org-acme", false),
+				newV1alpha1App("3def4-cluster-autoscaler", "cluster-autoscaler", "3def4", "org-acme", false),
+				newV1alpha1App("3def4-kiam", "kiam", "3def4", "org-acme", false),
+			},
+			charts: []runtime.Object{
+				newV1alpha1Chart("kiam", "giantswarm"),
+			},
+			expectAnnotation: map[string]bool{
+				"1abc2-app-operator":       false,
+				"1abc2-cert-exporter":      true,
+				"1abc2-cert-operator":      true,
+				"1abc2-cluster-autoscaler": true,
+				"1abc2-kiam":               false,
+				// different cluster
+				"3def4-app-operator":       false,
+				"3def4-cert-exporter":      false,
+				"3def4-cert-operator":      false,
+				"3def4-cluster-autoscaler": false,
+				"3def4-kiam":               false,
+			},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d: %s", i, tc.name), func(t *testing.T) {
+			var err error
+
+			cr, err := key.ToApp(tc.chartoperator)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			schemeBuilder := runtime.SchemeBuilder{
+				v1alpha1.AddToScheme,
+			}
+
+			err = schemeBuilder.AddToScheme(scheme.Scheme)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			fakeLogger := microloggertest.New()
+
+			var r *Resource
+			{
+				objs := tc.apps
+				objs = append(objs, tc.chartoperator)
+
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(objs...).
+					Build()
+
+				fakeK8sClient := clientgofake.NewSimpleClientset()
+
+				var valuesService *values.Values
+				{
+					c := values.Config{
+						K8sClient: fakeK8sClient,
+						Logger:    fakeLogger,
+					}
+
+					valuesService, err = values.New(c)
+					if err != nil {
+						t.Fatalf("error == %#v, want nil", err)
+					}
+				}
+
+				c := Config{
+					CtrlClient: fakeCtrlClient,
+					FileSystem: afero.NewMemMapFs(),
+					K8sClient:  fakeK8sClient,
+					Logger:     fakeLogger,
+
+					ChartNamespace:    "giantswarm",
+					WorkloadClusterID: "1abc2",
+					Values:            valuesService,
+				}
+
+				r, err = New(c)
+				if err != nil {
+					t.Fatalf("error == %#v, want nil", err)
+				}
+			}
+
+			var ctx context.Context
+			{
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(tc.charts...).
+					Build()
+
+				fakeClient := k8sclienttest.NewClients(k8sclienttest.ClientsConfig{
+					CtrlClient: fakeCtrlClient,
+					K8sClient:  clientgofake.NewSimpleClientset(),
+				})
+
+				defaultContent := helmclient.ReleaseContent{
```

Comment:
This will skip certain part of the `EnsureCreated` where this release is actually installed, though I don't believe it's necessarily bad since we are interested in the end result of having `StatusDeployed`, so that Apps can be annotated. And whether it is truly installed or not does not seem to depend on code.